tools/autograd/gen_variable_type.py

```diff
@@ -209,6 +209,14 @@
 RECORD_FUNCTION = CodeTemplate("""\
 profiler::RecordFunction profiler("${name}", Function::peek_at_next_sequence_nr());""")
 
+RECORD_FUNCTION_WITH_INPUTS = CodeTemplate("""\
+profiler::RecordFunction profiler("${name}", Function::peek_at_next_sequence_nr(), [${capture_inputs}]() {
+  jit::Stack inputs;
```

Comment:
Pretty sure this entire statement can just be 
```
[&] -> Stack { return {${names}}; }
```
Where names is just `[
            arg['name'] for arg in declaration['arguments']
            if check_type_for_capture(arg['simple_type'])]`